### PR TITLE
Upload functionality

### PR DESCRIPTION
**Pull request by fadezak100**

### CODE REVIEW

File: client/src/components/AddPresentation/index.tsx [hunk 1]
```diff
@@ -1,11 +1,22 @@
+import { useState } from 'react'
+
 import { ReactComponent as NewPresentation } from '../../assets/PresentationIcons/addPresentation.svg'
+import { Upload } from '..'
```
Comment (AhmedSafi97) [inline client/src/components/AddPresentation/index.tsx:4]: :eyes: what is this import path!

File: client/src/components/Upload/index.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { createRef } from 'react'
+
+export default function Modal() {
+  const toggled = createRef<HTMLDivElement>()
+
+  const handleToggleBody = () => {
+    console.log('ff')
```
Comment (AhmedSafi97) [inline client/src/components/Upload/index.tsx:7]: so we will just leave it with a console.log! if you want to do anything later then please leave a comment that explains so `// TODO: we still need to do so and so`

File: client/src/components/Upload/index.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  return (
+    <>
+      <>
```
Comment (AhmedSafi97) [inline client/src/components/Upload/index.tsx:12]: two nested fragments! what for?

File: client/src/components/index.ts [hunk 4]
```diff
@@ -6,3 +6,4 @@ export { default as Signup } from './Signup'
 export { default as Login } from './Login'
 export { default as PresentationCard } from './PresentationCard'
 export { default as AddPresentation } from './AddPresentation'
+export { default as Upload } from './Upload'
```
Comment (AhmedSafi97) [inline client/src/components/index.ts:9]: upload what exactly!

File: src/config/cloudinary.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import environment from './environment'
+
+const cloudinaryConfig = {
+  secure: true,
+  cloud_name: environment.cloudinary.cloudName,
+  api_key: environment.cloudinary.apiKey,
+  api_secret: environment.cloudinary.apiSecret
+}
+
+export default cloudinaryConfig
```
Comment (AhmedSafi97) [inline src/config/cloudinary.ts:10]: nice move :heart: !

File: src/config/environment.ts [hunk 6]
```diff
@@ -6,19 +6,37 @@ import dotenv from 'dotenv'
...
   },
   jwt: {
     secretKey: SECRET_KEY || ''
+  },
+  cloudinary: {
+    cloudName: CLOUD_NAME,
+    apiKey: CLOUDINARY_API_KEY,
+    apiSecret: CLOUDINARY_API_SECRET
+  },
+  readonly: {
```
Comment (AhmedSafi97) [inline src/config/environment.ts:38]: hmm readonly refers to what exactly?

File: src/config/rebrandly.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import environment from './environment'
+
+const rebrandlyConfig = {
+  'Content-Type': 'application/json',
+  apikey: environment.readonly.apiKey
+}
+
+export default rebrandlyConfig
```
Comment (AhmedSafi97) [inline src/config/rebrandly.ts:8]: Fuyoh!

File: src/controllers/SlideController.ts [hunk 8]
```diff
@@ -1,10 +1,46 @@
...
+const uploadSlide = async (
+  req: FileUploadRequest,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const file = req.file
+    const { id } = res.locals.user
+
+    const secureLink = await SlideService.uploadSlide(file)
```
Comment (AhmedSafi97) [inline src/controllers/SlideController.ts:21]: naming

File: src/controllers/SlideController.ts [hunk 9]
```diff
@@ -1,10 +1,46 @@
...
+    const validateShortLink = await validator({
+      schema: shortenLinkSchema,
+      data: { shortenLink }
+    })
+
+    if (!validateShortLink.isValid) {
+      throw new GenericError(validateShortLink.error)
+    }
+
+    await SlideService.addSlideToUser(id, shortenLink)
```
Comment (AhmedSafi97) [inline src/controllers/SlideController.ts:34]: createSlide!

File: src/controllers/SlideController.ts [hunk 10]
```diff
@@ -1,10 +1,46 @@
...
+
+    res.status(200).send({ message: 'success' })
```
Comment (AhmedSafi97) [inline src/controllers/SlideController.ts:36]: 201 since you have created a new resource

File: src/controllers/SlideController.ts [hunk 11]
```diff
@@ -1,10 +1,46 @@
...
+  } catch (error: unknown) {
+    const exception = error as Error
+
+    if (exception.name !== 'GenericError') return next(exception)
+    res.status(400).json({ message: exception.message })
+  }
+}
 
 const createSlide = async (
```
Comment (AhmedSafi97) [inline src/controllers/SlideController.ts:45]: well no we have two controllers missed up what is the difference between uploadSlide & createSlide? :thinking: 

I think it should squashed into one!

File: src/controllers/SlideController.ts [hunk 12]
```diff
@@ -40,7 +76,7 @@ const createSlide = async (
 }
 
 const deletePresentation = async (
-  req: DeleteSlideReqeust,
+  req: DeleteSlideRequest,
```
Comment (AhmedSafi97) [inline src/controllers/SlideController.ts:79]: nice catch

File: src/helpers/__tests__/SlideHelper.test.ts [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+beforeAll(() => {
+  return dbConnection()
+})
+
+afterAll(() => {
+  return dbConnection().then(db => db.connection.close())
+})
+
+//shortenLink
```
Comment (AhmedSafi97) [inline src/helpers/__tests__/SlideHelper.test.ts:12]: what is this comment telling me?

File: src/repositories/SlideRepository.ts [hunk 14]
```diff
@@ -1,8 +1,16 @@
 import Slide from '../models/SlideModel'
 import { CreateSlideOptions } from '../interfaces/SlideInterface'
+import User from '../models/UserModel'
 
 const createSlide = (slide: CreateSlideOptions) => Slide.create({ ...slide })
 
 const deleteSlide = (id: string) => Slide.findByIdAndDelete(id)
 
-export default { deleteSlide, createSlide }
+const addSlideToUser = (slide: CreateSlideOptions, id: string) =>
```
Comment (AhmedSafi97) [inline src/repositories/SlideRepository.ts:9]: naming :man_shrugging:

**Review by AhmedSafi97 — CHANGES_REQUESTED** [review]

File: client/src/components/AddPresentation/index.tsx (hunk 1 above)
Comment (fadezak100) [reply client/src/components/AddPresentation/index.tsx:4]: importing from the index of the previous file 
something like `../index`

File: src/config/environment.ts (hunk 6 above)
Comment (fadezak100) [reply src/config/environment.ts:38]: typo

File: src/config/rebrandly.ts (hunk 7 above)
Comment (fadezak100) [reply src/config/rebrandly.ts:8]: تحياتي

**Review by AhmedSafi97 — APPROVED** [review]

**Pull request merged**